### ISSUE TITLE
Roboform extension not working in sandboxed Firefox 106

### ISSUE THREAD

**Issue by darchon7** [sandboxie-plus/Sandboxie#2411]
### Describe what you noticed and did

click on the roboform icon at the top bar. an error box immediately appears saying "Can not start Roboform"

![rf-chrome-nm-host_axLF14pkWV](https://user-images.githubusercontent.com/14902121/198516228-4328c8cd-bda9-4370-ad23-8cbc8aadf31c.png)

works outside the sandbox and in previous versions.

other extensions seem to be working fine.

### How often did you encounter it so far?

always

### Affected program

Firefox portable 106..0.2

### Download link

https://addons.mozilla.org/en-US/firefox/addon/roboform/?utm_source=addons.mozilla.org&utm_medium=referral&utm_content=search

### Where is the program located?

The program is installed only outside the sandbox.

### Expected behavior

a roboform popup window should appear asking to login which then provides access to passwords stored in D:\My Documents\RoboForm

### What is your Windows edition and version?

windows 10 pro 21h2 64-bit b19044.2130

### In which Windows account you have this problem?

User account with UAC protection completely turned off.

### Please mention any installed security software

Bitdefender 

### What version of Sandboxie are you running?

Sandboxie Plus 1.5.1

### Is it a new installation of Sandboxie?

I just updated Sandboxie from a previous version (to be specified).

### In which sandbox type you have this problem?

In a standard isolation sandbox (yellow sandbox icon).

### Can you reproduce this problem on an empty sandbox?

I can confirm it also on an empty sandbox.

### Did you previously enable some security policy settings outside Sandboxie?

no

### Sandboxie.ini configuration

```shell
[GlobalSettings]

Template=Edge_Fix
Template=RpcPortBindings
Template=WindowsRasMan
Template=BitDefenderInternetSecurity
Template=7zipShellEx
Template=Listary
Template=WindowsLive
Template=OfficeLicensing
Template=OfficeClickToRun
Template=RoboForm
FileRootPath=d:\Sandbox\%USER%\%SANDBOX%
KnownConflicts=BitDefender
KeyRootPath=\REGISTRY\USER\Sandbox_%USER%_%SANDBOX%
IpcRootPath=\Sandbox\%USER%\%SANDBOX%\Session_%SESSION%

[DefaultBox]

ConfigLevel=9
AutoRecover=y
BlockNetworkFiles=n
Template=OpenBluetooth
Template=SkipHook
Template=FileCopy
Template=qWave
Template=BlockPorts
Template=LingerPrograms
Template=Chrome_Phishing_DirectAccess
Template=Firefox_Phishing_DirectAccess
Template=AutoRecoverIgnore
RecoverFolder=%{374DE290-123F-4565-9164-39C4925E467B}%
RecoverFolder=%Personal%
RecoverFolder=%Favorites%
RecoverFolder=%Desktop%
RecoverFolder=d:\Program Files
RecoverFolder=F:\Downloads
RecoverFolder=C:\Users\dwayne\Downloads
RecoverFolder=C:\Users\dwayne\AppData\Local\RoboForm
BorderColor=#00ffff,ttl,6
Enabled=y
OpenFilePath=<firefoxprograms>,D:\Program Files\
OpenFilePath=firefoxportable.exe,D:\Program Files\
OpenFilePath=firefox.exe,D:\Program Files\
OpenFilePath=<firefoxprograms>,F:\Downloads\
OpenFilePath=firefoxportable.exe,F:\Downloads\
OpenFilePath=firefox.exe,F:\Downloads\
OpenFilePath=robotaskbaricon.exe,%Personal%\My RoboForm Data\
OpenFilePath=robotaskbaricon.exe,C:\Program Files (x86)\Siber Systems\AI RoboForm\
OpenFilePath=qbittorrent.exe,%AppData%\qBittorrent\
OpenFilePath=qbittorrent.exe,%Local AppData%\qBittorrent\
OpenFilePath=qbittorrent.exe,H:\
OpenFilePath=operaportable.exe,D:\Program Files\
OpenFilePath=opera.exe,D:\Program Files\
OpenFilePath=chrome.exe,D:\Program Files\
OpenFilePath=googlechromeportable.exe,D:\Program Files\
OpenFilePath=rf-chrome-nm-host.exe,C:\Users\dwayne\AppData\Local\RoboForm\
OpenPipePath=qbittorrent.exe,%{374DE290-123F-4565-9164-39C4925E467B}%\
ClosedFilePath=*BitDefender*atcuf*.dll
WriteFilePath=C:\Program Files\Bitdefender\*
OpenWinClass=#
BoxNameTitle=n
BlockNetParam=y
DropAdminRights=n
FakeAdminRights=n
ClosePrintSpooler=n
OpenPrintSpooler=n
AllowSpoolerPrintToFile=n
CopyLimitKb=81920
CopyLimitSilent=n
NeverDelete=n
AutoDelete=n
AllowRawDiskRead=n
NotifyDirectDiskAccess=n

[UserSettings_11C20289]

SbieCtrl_UserName=dwayne
SbieCtrl_BoxExpandedView=DefaultBox
SbieCtrl_NextUpdateCheck=1614796928
SbieCtrl_WindowCoords=200,150,909,596
SbieCtrl_ActiveView=40021
SbieCtrl_UpdateCheckNotify=n
SbieCtrl_AutoApplySettings=n
SbieCtrl_ProcessViewColumnWidths=250,70,300
SbieCtrl_AutoStartAgent=SandMan.exe
SbieCtrl_EnableAutoStart=y

[UserSettings]

SbieCtrl_AutoStartAgent=SandMan.exe
AutoRecover=n
BlockNetworkFiles=y
RecoverFolder=%{374DE290-123F-4565-9164-39C4925E467B}%
RecoverFolder=%Personal%
RecoverFolder=%Desktop%
BorderColor=#00FFFF,ttl
Template=OpenBluetooth
Template=SkipHook
Template=FileCopy
Template=qWave
Template=BlockPorts
Template=LingerPrograms
Template=Chrome_Phishing_DirectAccess
Template=Firefox_Phishing_DirectAccess
Template=AutoRecoverIgnore
ConfigLevel=9
```

**Comment by offhub** [comment]
Try adding this to see if it resolves the problem.
`ReadFilePath=firefox.exe,*\Program Files (x86)\Siber Systems\AI RoboForm\*`

**Comment by darchon7** [comment]
THANKS A BUNCH!! that fixed it.

**Comment by isaak654** [comment]
It may be worthwhile to refine the existing template before closing the issue:
https://github.com/sandboxie-plus/Sandboxie/blob/65f936b3a4f7d041cf42484287ed5fe83a619923/Sandboxie/install/Templates.ini#L2275-L2285

**Comment by DavidXanatos** [comment]
PLease clarify has this issue been caused by a sandboxie update or by a firefox update?

**Comment by offhub** [comment]
Started with Firefox 106 update.

**Comment by isaak654** [comment]
The workaround doesn't seem to work with [custom paths](https://www.wilderssecurity.com/threads/sandboxie-plus-v1-5-0-v1-5-1-v1-5-2.448352/page-5#post-3114549), so it would probably need to be refined a bit:

> Thanks a lot "APMichael" for pointing out this already ongoing discussion about that topic.
> 
> Unfortunately the "workaround"-line given there didn't do the trick with my installation. Reason probably is that
> ReadFilePath=firefox.exe,*\Program Files (x86)\Siber Systems\AI RoboForm\*,
> - while it exists on my system - is an empty directory.
> 
> My Roboform is installed in C:\WINTOOL\Roboform\ which again has no "Siber Systems" or "AI RoboForm" SubDir. So I've done a system-wide search with the result that an "AI RoboForm"-SubDir, in particular with any content, doesn't exist ANYWHERE ELSE in my Dir-structure either to point a "ReadFilePath="-statement to.
> 
> Unfortunately adding a more general
> ReadFilePath=firefox.exe,*\WINTOOL\RoboForm\*
> statement didn't help in remedying the issue either. So I'm not sure about that "workaround" but thanks again for pointing to that github-thread anyway.

**Comment by offhub** [comment]
I tested with installing to that folder and it works for me. Something else should be causing the strange behavior.

Firefox 106.0.5 64-bit
RoboForm Password Manager 9.3.5.0 (Firefox)
RoboForm 9.3.7.7 (Windows)
Windows 10 22H2

Install location
`C:\WINTOOL\Roboform\`

Sandboxie.ini
`ReadFilePath=firefox.exe,*\WINTOOL\RoboForm*`

**Comment by DavidXanatos** [comment]
where to find a download of that? on the website there is only a firefox extension which does not install any external components

**Comment by isaak654** [comment]
> where to find a download of that? on the website there is only a firefox extension which does not install any external components

https://www.roboform.com/dist/RoboForm-v9-Setup.exe

![roboform_for_Windows](https://user-images.githubusercontent.com/12372772/200298185-be2a0fa5-2df4-484c-8b08-d7d09509e489.png)

**Comment by DavidXanatos** [comment]
the issue is properly fixed it was the same issue as #2391
